jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -125,87 +125,94 @@ ecma_regexp_parse_flags (ecma_string_t *flags_str_p, /**< Input string with flag
-                                   NULL);
-  ecma_create_named_data_property (re_object_p,
-                                   ecma_get_magic_string (LIT_MAGIC_STRING_IGNORECASE_UL),
-                                   ECMA_PROPERTY_FIXED,
-                                   NULL);
-  ecma_create_named_data_property (re_object_p,
-                                   ecma_get_magic_string (LIT_MAGIC_STRING_MULTILINE),
-                                   ECMA_PROPERTY_FIXED,
-                                   NULL);
-#endif /* !ENABLED (JERRY_ES2015) */
-  ecma_create_named_data_property (re_object_p,
-                                   ecma_get_magic_string (LIT_MAGIC_STRING_LASTINDEX_UL),
-                                   ECMA_PROPERTY_FLAG_WRITABLE,
-                                   NULL);
+  ecma_property_value_t *prop_value_p;
+
+  prop_value_p = ecma_create_named_data_property (re_object_p,
+                                                  ecma_get_magic_string (LIT_MAGIC_STRING_SOURCE),
+                                                  ECMA_PROPERTY_FIXED,
+                                                  NULL);
+
+  ecma_ref_ecma_string (source_p);
+  prop_value_p->value = ecma_make_string_value (source_p);
+
+  prop_value_p = ecma_create_named_data_property (re_object_p,
+                                                  ecma_get_magic_string (LIT_MAGIC_STRING_GLOBAL),
+                                                  ECMA_PROPERTY_FIXED,
+                                                  NULL);
+
+  prop_value_p->value = ecma_make_boolean_value (flags & RE_FLAG_GLOBAL);
+
+  prop_value_p = ecma_create_named_data_property (re_object_p,
+                                                  ecma_get_magic_string (LIT_MAGIC_STRING_IGNORECASE_UL),
+                                                  ECMA_PROPERTY_FIXED,
+                                                  NULL);
+
+  prop_value_p->value = ecma_make_boolean_value (flags & RE_FLAG_IGNORE_CASE);
+
+  prop_value_p = ecma_create_named_data_property (re_object_p,
+                                                  ecma_get_magic_string (LIT_MAGIC_STRING_MULTILINE),
+                                                  ECMA_PROPERTY_FIXED,
+                                                  NULL);
+
+  prop_value_p->value = ecma_make_boolean_value (flags & RE_FLAG_MULTILINE);
 } /* ecma_regexp_create_props */
 
 /*
- * Helper function to assign a value to a property
+ * Update the properties of a RegExp instance.
  */
 static void
-ecma_regexp_helper_assign_prop (ecma_object_t *re_object_p, /**< RegExp object */
-                                lit_magic_string_id_t prop_id, /**< property name ide */
-                                ecma_value_t value) /**< value */
+ecma_regexp_update_props (ecma_object_t *re_object_p, /**< RegExp object */
+                          ecma_string_t *source_p, /**< source string */
+                          uint16_t flags) /**< flags */
```

Comment:
Could you group the code in this function a bit logically? It's not very nice to read.